src/main/java/subway/LineService.java

```diff
@@ -35,16 +40,50 @@ private LineResponse createLineResponse(Line line, Station upStation , Station d
                 List.of(new StationResponse(upStation.getId(), upStation.getName()), new StationResponse(downStation.getId(), downStation.getName())));
     }
 
+    @Transactional(readOnly = true)
     public List<LineResponse> findLines() {
         List<Line> lines = lineRepository.findAll();
         return lines.stream().map(this::createLineResponse).collect(Collectors.toList());
     }
 
+    @Transactional(readOnly = true)
     public LineResponse findLine(Long id) {
         Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException("해당 아이디의 지하철 노선이 없습니다."));
         return createLineResponse(line);
     }
 
+    @Transactional
+    public void updateLine(Long id, LineRequest request) {
+        Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException("해당 아이디의 지하철 노선이 없습니다."));
+        line.updateLine(request.getName(), request.getColor());
+    }
+
+    @Transactional
+    public void deleteLine(Long id) {
+        lineRepository.deleteById(id);
+    }
+
+    @Transactional
+    public LineResponse addLine(Long id, AddLineRequest dto) {
+        Station upStation = stationRepository.findById(dto.getUpStationId()).orElseThrow(() -> new RuntimeException("상행선역 지하철을 먼저 등록해주세요"));
+        List<LineStation> lines = lineStationRepository.findAllByLineIdOrderBySequenceDesc(id);
+        Station prevDownStation = lines.stream().findFirst().orElseThrow(() -> new RuntimeException("해당하는 지하철 노선 id가 없습니다.")).getStation();
+        dto.compareUpStationId(prevDownStation.getId());
+        LineStation lineStation = lines.stream()
+                .max(Comparator.comparing(LineStation::getSequence))
+                .orElseThrow(() -> new RuntimeException("해당하는 지하철 노선의 max sequence 값이 없습니다."));
+        LineStation save = lineStationRepository.save(new LineStation(lineStation.getSequence() + dto.getDistance() - 1,
+                lineStation.getLine(),
+                upStation));
+        return createLineResponse(save.getLine());
```

Comment:
구간을 추가하는 비즈니스 로직을 최대한 도메인 영역으로 옮겨보면 어떨까요?

Line은 LineStation 목록을 가지고 있기 때문에 Line객체만 조회하면 Line과 관련되어있는 LineStation 목록은 함께 조회 됩니다.
그렇다면 Station과 Line을 조회하는 로직 이외에는 모두 LineService 외 다른 객체로 위임이 가능해 보입니다.

고로 궁극적으로는 이러한 형태를 띌 수 있을 것 같습니다.

```suggestion
        Station upStation = stationRepository.findById(dto.getUpStationId()).orElseThrow(() -> new RuntimeException("상행선역 지하철을 먼저 등록해주세요"));
        Line line = lineRepository.findById(id).orElseThrow(() -> new RuntimeException());
        line.addSection(~~~);
        return createLineResponse(line);
```